### PR TITLE
Fix format on paste

### LINKED ISSUE #51: Don't refmt a selection if it's just a part of a line
Hi, not sure if thats intended but the extension adds a semicolon to whatever I paste in the editor. That might be helpful in a more context-sensitive way but right now it is a lot of work removing all those semicolons.

Here is a small example:
semicolon_bug

Version is the latest Beta 5!

### PR DESCRIPTION
**Pull request by no-stack-dub-sack**
Thanks for the help @thangngoc89  :-) 

closes #51

### CODE REVIEW

File: src/lsp/MessageHandlers.re [hunk 1]
```diff
@@ -333,78 +333,92 @@ let handlers: list((string, (state, Json.t) => result((state, Json.t), string)))
...
-          let line = List.nth(splitted, i);
-          switch (line |> String.trim |> String.length) {
-          | 0 => loop(i + 1, leadingLines + 1, skipChars + (line |> String.length))
-          | _ => (leadingLines, skipChars + 1)
+      /* 
+        This is to prevent formatOnPaste 
+        See https://github.com/jaredly/reason-language-server/issues/51 
+        for more information
+      */
+      if (fst(start) == fst(end_) && text.[endPos] == '\n') {
```
Comment (jaredly) [inline src/lsp/MessageHandlers.re]: hmmm did you check that this works? I would think you'd want `text.[endPos] != '\n'`

Comment (thangngoc89) [reply src/lsp/MessageHandlers.re]: I believe this should be `text.[endPos] == '\n'`. Or just skip this condition all together. 
Reasons:

- You can copy/paste multiple lines
- For a single line, it stills add a new semicolon after pasted content if it's `!= '\n'` 


I mean this is a really annoying "feature" of VSCode. It should give us a head up about formatOnPaste or formatOnSave

Comment (jaredly) [reply src/lsp/MessageHandlers.re]: Here's my desired outcome:

- when I select a whole line and say "format selection", it should refmt it
- when someone pasts in a subsection of a line, it should not be refmt'd
- anything multiline is fair game for refmt'ing. If you have "format on paste" turned on, then that's what you get.

Comment (thangngoc89) [reply src/lsp/MessageHandlers.re]: Alright. I think I misunderstood the desired behaviour and totally forgot about rangeFormating  the a single line. Fixed it

**Comment by jaredly** [comment]
oops, now there are conflicts :/

**Comment by jaredly** [comment]
manually merged in. Thanks!

**Pull request merged**